File: backend/internal/model/inventory.go
package model

import "time"

// Inventory 库存记录（按商品×批次×库位）。
type Inventory struct {
	ID            uint      `gorm:"primaryKey" json:"id"`
	ProductID     uint      `gorm:"index:idx_inv_prod_bin,unique;not null" json:"product_id"`
	OwnerID       uint      `gorm:"index;not null" json:"owner_id"`
	BinLocationID uint      `gorm:"index:idx_inv_prod_bin,unique;not null" json:"bin_location_id"`
	BatchNo       string    `gorm:"size:64;index:idx_inv_prod_bin,unique" json:"batch_no"`
	Quantity      int       `gorm:"not null" json:"quantity"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

File: frontend/src/api/inventory.ts
import { get } from '../utils/request'
import type { Paginated, PageQuery } from '../types/enums'

export interface InventoryView {
  id: number
  product_id: number
  product_name: string
  sku: string
  owner_id: number
  owner_name: string
  bin_location_id: number
  bin_code: string
  batch_no: string
  quantity: number
  unit_price: number
  total_value: number
  updated_at: string
}

export interface InventorySummaryItem {
  owner_id: number
  owner_name: string
  total_qty: number
  total_value: number
}

export function apiListInventory(params: PageQuery) {
  return get<Paginated<InventoryView>>('/inventory', params)
}

export function apiInventorySummary() {
  return get<InventorySummaryItem[]>('/inventory/summary')
}
